### ISSUE TITLE
clangd 20.1.1 cannot locate sys/features.h in ESP-IDF 6.0.2, causing false diagnostics (memset, ESP_LOGI, etc.)

### ISSUE THREAD

**Issue by freitasj1** [espressif/llvm-project#135]
### Checklist

- [x] Checked the issue tracker for similar issues to ensure this is not a duplicate
- [x] Read the documentation to confirm the issue is not addressed there and your configuration is set correctly
- [x] Tested with the latest version to ensure the issue hasn't been fixed

### How often does this bug occurs?

always

### Expected behavior

When opening an ESP-IDF project with clangd, standard library headers should be resolved correctly.

`<string.h>` should be parsed successfully, functions such as `memset()` should be recognized, and ESP-IDF macros such as `ESP_LOGI()` should not produce diagnostics.

The project builds successfully with `idf.py build`, so clangd is expected to report no syntax errors.

### Actual behavior (suspected bug)

clangd reports false diagnostics in valid ESP-IDF source files.

Examples include:

- `Use of undeclared identifier 'memset'`
- `Use of undeclared identifier 'memcpy'`
- `Expected ')'` on every `ESP_LOGI(...)`

Running clangd in check mode shows that the first error is:

```
'sys/features.h' file not found
```

All subsequent diagnostics appear to be consequences of this missing header.

The project compiles successfully with `idf.py build`.

### Error logs or terminal output

```shell
The complete output of the following command is attached as `clangd.log`:


/home/<user>/.espressif/tools/esp-clang/esp-20.1.1_20250829/esp-clang/bin/clangd \
  --check=components/dsp_pipeline/dsp.c \
  --compile-commands-dir=build


Relevant excerpts:


Loaded compilation database from build/compile_commands.json

Built preamble...

unknown type name '__FILE'

All checks completed, 6 errors
```

### Steps to reproduce the behavior

1. Install ESP-IDF v6.0.2.
2. Install Espressif clangd 20.1.1.
3. Open any ESP-IDF project in VS Code.
4. Build the project successfully using:

```bash
idf.py build
```

5. Open a source file that includes `<string.h>` or uses `ESP_LOGI()`.

clangd reports false diagnostics even though the project builds successfully.

The issue can also be reproduced by running:

```bash
clangd --check=components/dsp_pipeline/dsp.c --compile-commands-dir=build
```

### Project release version

ESP-IDF v6.0.2

### System architecture

Intel/AMD 64-bit (modern PC, older Mac)

### Operating system

Linux

### Operating system version

Ubuntu 24.04 LTS

### Shell

ZSH

### Additional context

The project builds successfully with `idf.py build` and produces no compiler errors.

I investigated the issue using:

```bash
clangd --check=components/dsp_pipeline/dsp.c --compile-commands-dir=build
```

The compilation database is loaded correctly, but the first reported error is:

```
'sys/features.h' file not found
```

This appears to trigger a cascade of false diagnostics, including:

- `Use of undeclared identifier 'memset'`
- `Use of undeclared identifier 'memcpy'`
- `Expected ')'` on every `ESP_LOGI(...)`

I verified that `sys/features.h` is present in the installed toolchain, for example:

```
.../xtensa-esp-elf/xtensa-esp-elf/sys-include/sys/features.h
```

and

```
.../xtensa-esp-elf/picolibc/xtensa-esp-elf/sys-include/sys/features.h
```

As a workaround, I created a `.clangd` file and manually added the toolchain system include directories:

```yaml
CompileFlags:
  CompilationDatabase: build/
  Add:
    - -isystem
    - <toolchain>/xtensa-esp-elf/include
    - -isystem
    - <toolchain>/xtensa-esp-elf/sys-include
```

After adding these directories:

- `<string.h>` is resolved correctly;
- `memset()` and `memcpy()` are recognized;
- `ESP_LOGI()` is parsed correctly;
- all false diagnostics disappear.

This suggests that clangd may not be automatically adding the toolchain system include directories required by the ESP-IDF toolchain.

[clangd.log](https://github.com/user-attachments/files/30680298/clangd.log)

**Comment by antmak** [comment]
hi @freitasj1 ,

Yeah it's related to configuring clangd for cross-compilation toolchains. It can get a little tricky.

Actually the short version is:

_Сlangd config_:
```yaml
CompileFlags:
  Remove: [-m*, -f*]
  BuiltinHeaders: QueryDriver
```

_Running clangd from the editor_:
```
<path_to_clangd>/clangd --query-driver=**
```

These settings are all that clangd requires.

You can add `--log=verbose` to get more details, especially if the errors occur in some files but not in others.

I'd also recommend placing a `.clangd` file in every directory where you browse IDF-related or Espressif-specific code, such as your project dir and the esp-idf dir. (Alternatively, you can use `~/.config/clangd/config.yaml` as a global config, adding `If: { PathMatch: [ ] }` sections as needed)
